packages/buttons/src/elements/ToggleButton.tsx

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button, { IButtonProps } from './Button';
+
+interface IToggleButtonProps extends IButtonProps {
+  /**
+   * Determine if the button is pressed or not. Use `'mixed'` to indicate
+   * whether the toggle controls other elements which do not share the same
+   * value.
+   */
+  isPressed?: boolean | 'mixed';
+}
+
+/**
+ * A `Button` with the [ARIA
+ * attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#Toggle_buttons)
+ * to indicate a pressed state.
+ */
+const ToggleButton: React.FunctionComponent<
```

Comment:
Since toggle buttons are buttons with `aria-pressed` state, what do you think about adding `aria-pressed={isPressed}` directly to the `Button` component and removing this component (`ToggleButton`)? 

So when a user wants to use a toggle button the API would be:

```jsx
<Button isPressed={isPressed} />
```

Reply:
@hzhu, adding `isPressed` to `Button` tends to have the unintended affect of turning all Garden buttons into toggle buttons. Due to the important accessibility distinction, I elected to make `ToggleButton` its own component.

The most recent deployment will have a dedicated `ToggleButton` example. Please have a look at that and the linked documentation (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#Toggle_buttons), turn on VO and observe the difference between how a `Button` is read vs. a `ToggleButton`.

Reply:
@jzempel how would adding `isPressed` make all buttons register as toggle buttons?

The default case of `<Button>foo</Button>` would have the value of `<button aria-pressed={undefined}>foo</button>`.

[I don't believe that would register as a toggle button.](https://www.w3.org/TR/wai-aria-1.1/#aria-pressed)

Reply:
@austin94 true, but now you've encountered the flip side of the problem. That is that a consumer would need to understand the difference between `<Button>` and `<Button isPressed={false}>` in order to provide a successful toggle button.

Given the fact that a toggle button is not a flavor variety, but a fundamentally different type of button, I think that providing a `<ToggleButton>` is a more human-centered API.